src/index.js

```diff
@@ -274,6 +274,34 @@ class PublicationClient extends EventEmitter {
     return subscription;
   }
 
+  /**
+   * Subscribe to the publication with the given name. Any other parameters
+   * are passed as arguments to the publication.
+   *
+   * @param {String} name The publication to subscribe to.
+   * @param {Object} options containing optional parameters such as bootstrap and/or
+   * bootstrapAsync.
+   * @param {*[]} params (optional) Params to pass to the publication.
+   * @returns {Subscription} The subscription to the desired publication.
+   */
+  subscribeWithOptions(name, options, ...params) {
+    const id = this._nextSubscriptionId++;
+    const payload = [...params, options];
+
+    // Hash the name and params to cache the subscription including payload extended with options.
+    const subscriptionKey = JSON.stringify([name].concat(payload));
+    var subscription = this._subscriptions[subscriptionKey];
```

Comment:
It feels like while we're here we ought to update to our latest code standard of preferring `let` to `var`. Are you aware of any compatibility or other restrictions in this module that would prevent us from using `let`?